Source/Core/VideoCommon/HiresTextures.cpp

```diff
@@ -144,22 +190,81 @@ void HiresTexture::Prefetch()
+        (Config::Get(Config::GFX_HIRESTEX_OVERRIDE_PREFETCH_CULLED_PERCENT) > 90.0))
+    {
+      if (Config::Get(Config::GFX_HIRESTEX_OVERRIDE_PREFETCH_CULLED_PERCENT) < 5.0)
+      {
+        percent_mem_to_cull = 5.0;
+        OSD::AddMessage(
+            "Prefetcher: User INI setting for 'culled cercent' of " +
+                StringFromFormat(
+                    "%.1f %%", Config::Get(Config::GFX_HIRESTEX_OVERRIDE_PREFETCH_CULLED_PERCENT)) +
+                " is below the lowest bound. Resetting to lowest allowable value.",
+            20000);
+        WARN_LOG(VIDEO, "Prefetcher: Mem limit cull percent INI override out of "
+                        "lower bound. Setting to lowest");
+      }
+      else if (Config::Get(Config::GFX_HIRESTEX_OVERRIDE_PREFETCH_CULLED_PERCENT) > 90.0)
+      {
+        percent_mem_to_cull = 90.0;
+        OSD::AddMessage(
+            "Prefetcher: User INI setting for 'culled percent' of " +
+                StringFromFormat(
+                    "%.1f %%", Config::Get(Config::GFX_HIRESTEX_OVERRIDE_PREFETCH_CULLED_PERCENT)) +
+                " is below the highest bound. Resetting to highest allowable value.",
+            20000);
+        WARN_LOG(VIDEO, "Prefetcher: Mem limit cull percent INI override out of "
+                        "higher bound. Setting to highest");
+      }
+    }
+    else
+    {
+      percent_mem_to_cull = Config::Get(Config::GFX_HIRESTEX_OVERRIDE_PREFETCH_CULLED_PERCENT);
+      INFO_LOG(VIDEO, "Prefetcher: Mem limit overriden by user INI successfully, "
+                      "at %.2f (cull percent)",
+               percent_mem_to_cull);
+    }
+  }
+
+  double prefetched_size_sum = 0;  // total mem size incl. non-native decompressed textures
+  const double absolute_max_mem = Common::MemPhysical();  // excl. HW reserved
+  const double culled_amount = (Common::MemPhysical() * percent_mem_to_cull) / 100;
+  const double delegated_max_mem = Common::MemPhysical() - culled_amount;
+  const double below_min_req_mem = 2147483648;
+
+  if (absolute_max_mem <= below_min_req_mem)
+  {
+    OSD::AddMessage("Prefetcher: Warning! System RAM of " +
+                        StringFromFormat("%.2f MB", absolute_max_mem / (1024.0 * 1024.0)) +
+                        " is below Dolphin's minimum recommended specifications.",
+                    20000);
+    WARN_LOG(VIDEO, "Usable system memory below Dolphin's minimum recommended specifications");
+  }
+
+  // Start Prefetching
+  prefetch_start_time_total = Common::Timer::GetTimeMs();
+
+  for (const auto& texmap_ref : s_textureMap)
   {
-    const std::string& base_filename = entry.first;
+    const std::string& base_filename = texmap_ref.first;
 
     if (base_filename.find("_mip") == std::string::npos)
     {
       {
-        // try to get this mutex first, so the video thread is allow to get the real mutex faster
+        // prefetch() is running in a tight loop, and it's using that to allow
```

Comment:
That's because JonnyH was talking about mutexes when figuring things out - let's take a look again if the subject was changed or something else.

![gitcontrib_dolphin_customtextures_prefetcheroverhaulcomm2_mutexlocksirclogjonnyh](https://user-images.githubusercontent.com/20824154/27996231-21e049ba-64de-11e7-80c3-8226e630a93e.png)